cartographer/mapping/2d/grid_2d.cc

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "cartographer/mapping/2d/grid_2d.h"
+
+#include "cartographer/mapping/probability_values.h"
+
+namespace cartographer {
+namespace mapping {
+namespace {
+
+// Converts a 'cell_index' into an index into 'cells_'.
+int ToFlatIndex(const Eigen::Array2i& cell_index, const MapLimits& limits) {
```

Comment:
move it to Grid2D protected section. Currently it is in both grid_2d and in probability_grid_2d.